src/logging/logger.ts

```diff
@@ -1,29 +1,24 @@
-import { App } from 'electron';
-import logger from 'electron-log';
+import { App } from '@/app-globals';
+import logger, { LogFunctions, LogLevel } from 'electron-log'; // eslint-disable-line no-restricted-imports
 import fs from 'fs';
 import { EOL } from 'os';
 
-declare module 'electron-log' {
-  interface ElectronLog {
-    getLastLines: (n: number) => string
-  }
-}
+export type Logger = LogFunctions
+export type Levels = LogLevel
 
-export default function CreateLogger(app: App) {
-  logger.info(`Welcome to ${app.getName()} log`);
-  logger.info(`Version: ${app.getVersion()}`);
+logger.info(`Welcome to ${App.getName()} log`);
+logger.info(`Version: ${App.getVersion()}`);
 
-  const onError = (error: Error) => {
-    logger.error(error.message ? error.message : error);
-    if (error.stack) logger.debug(error.stack);
-  };
-  logger.catchErrors({ onError });
+const onError = (error: Error) => {
+  logger.error(error.message || error);
+  if (error.stack) logger.debug(error.stack);
+};
+logger.catchErrors({ onError });
```

Comment:
Note, this catching only the `UnhandledRejection`.

At least for the Vue part, we need to open a new issue to catch exceptions during the render process (I even can't say what exactly happened during such case. For in React, for example, the component tree will crash- if you're not using error boundary)

Reply:
#198